test/integration/test/philips_light_test.js

```diff
@@ -0,0 +1,99 @@
+// This agent refers to PORT where program is runninng.
+const spawn = require('child_process').spawn;
+const chakram = require('chakram'), expect = chakram.expect;
+const Config = require('config-js');
+
+var philipshue_server = require('../lib/philipsHue_server.js');
+var nupnp_server = require('../lib/nupnp_PhilipsHue.js');
+var upnp_server = require('../lib/upnp_PhilipsHue.js');
+var config = new Config('./lib/config/foxbox.js');
+
+describe("Initiate the connection with foxbox as Philips Hue hub",function(){
+  var credential = config.get('credential');
+  var key = credential['username']+":"+credential['password'];
+  var encoded_cred = new Buffer(key).toString('base64');
+  var header = config.get('header');
+  var lightinfo; // to store the light status received from foxbox
+  var FOXBOX_STARTUP_WAIT_TIME_IN_MS = 5000;
+  var foxbox_process;
+  before(function(done){
+    this.timeout(500000);
+
+    // Start the nupnp server
+    const hue_location = "localhost:" + config.get('philips_hue.port');
+    nupnp_server.start(config.get('nupnp_server.id'),hue_location,config.get('nupnp_server.port'));
+    
+    // Start the Philips Hue simulator, and set the #2 lightbulb status to false
+    philipshue_server.setup(config.get('philips_hue.port'));
+    philipshue_server.setlight(2,false);
+
+    // kick off the foxbox within 10 seconds
+    remove_sqlite = spawn('rm',['users_db.sqlite']);
+    foxbox_process = spawn('../../target/debug/foxbox', 
+      ['-c', 'philips_hue;nupnp_url;http://localhost:'+ config.get('nupnp_server.port')+'/']);
+    setTimeout(done, FOXBOX_STARTUP_WAIT_TIME_IN_MS);
+    
+  });
+  
+  after(function(done){
+    foxbox_process.kill();
+    done();
+  });
+  
+  it("create a new credential",function(){
+
+    return chakram.post(config.get('foxbox.url') + "/users/setup",credential)
+    .then(function(setupResp){
+      expect(setupResp).to.have.status(201);
+    });
+  });
+
+  it("login to foxbox",function(){
+    header['Authorization'] = 'Basic ' + encoded_cred;  // supply the credential used in previous test
+    
+    return  chakram.post(config.get('foxbox.url') + "/users/login",null,{'headers' : header})
+    .then(function(loginResp){
+      expect(loginResp).to.have.status(201);
+    });
+  });
+
+  describe ("once logged in", function () {
+    before(function() {
+      return chakram.post(config.get('foxbox.url') + "/users/login",null,{'headers' : header})
+      .then(function(resp){
+       header['Authorization'] = 'Bearer ' + resp.body.session_token;
+       chakram.setRequestDefaults({'headers': header});
```

Comment:
Hmm, should we?  we pretty much need the token throughout the session, and if we decide to use a new account, we have to supply a new token anyways - there isn't a case where we need to explictly clear the header, what do you think?